src/uglify/worker.js

```diff
@@ -7,7 +8,15 @@ module.exports = (options, callback) => {
     // eslint-disable-next-line no-new-func, no-param-reassign
     options = new Function(`'use strict'\nreturn ${options}`)();
 
-    callback(null, minify(options));
+    const result = minify(options);
+
+    // Communication channel employed by process.send() converts everything to JSON first.
+    // So custom object types (like Errors) can't possibly work, since object types aren't represented in JSON.
+    if (result.error) {
+      result.error = serializeError(result.error);
+    }
+
+    callback(null, result);
```

Comment:
```js
if (result.error) cb(result.error)

cb(null, result)
```

?

```js
const path = require('path')

const postcss = require('postcss')
const postcssrc = require('postcss-load-config')

const SyntaxError = require('./SyntaxError')
const WorkerError = require('../WorkerError')

const css = (task, ctx, cb) => {
  const { plugins } = ctx
  const { file, data, options } = task

  return postcss(plugins)
    .process(data, options)
    .then(({ css, map }) => {
      cb(null, { file, css, map })
    })
    .catch((err) => {
      err.name === 'CssSyntaxError'
        ? cb(new SyntaxError(err))
        : cb(new WorkerError(err))
    })
}

module.exports = css
```

This works so far for `css-webpack-plugin`

Reply:
Whats is specific to `options.parallel` here ?

Reply:
@michael-ciniawsky i think about this, but for consistently don't use `error` in callback. Do you think better is use `callback(err, result)`?

`options.parallel` need for `worker-farm` when you use more than 1 process

Reply:
> Do you think better is use callback(err, result)?

Yep :), but I still wonder why the `{Error}` is malformed nevertheless, maybe the problem lies elsewhere...

Reply:
@michael-ciniawsky problem when you `process.send()`, it is call `JSON.stringify` and send data to other process (main), but `JSON.stringify` eats error(s). Just run `JSON.stringify({ error: new Error('test') });`

Reply:
**index.js** (CSS Plugin)
```js
workers.run(stylesheets, { plugins }, (errors, assets) => {
  if (errors && errors.length > 0) {
    errors.forEach((err) => {
      compilation.errors.push(err.message)
    })
}
```

**console.log(errors)**
```
[ { CSS Plugin:

      (7:3) Syntax Error: Unclosed block

        5 |   background-image: url('/assets/file.jpg');
        6 |   /* background-image: url('./file.png'); */
      > 7 | } {
          |   ^
        8 |
        9 |

          at postcss.process.then.catch (/Users/Cini/Github/Webpack/plugins/css-webpack-plugin/src/lib/worker/workers/postcss.js:22:14)
          at <anonymous>
          at runMicrotasksCallback (internal/process/next_tick.js:115:5)
          at process._tickCallback (internal/process/next_tick.js:150:11) type: 'SyntaxError', '$error': '$error', name: 'CSS Plugin' },
        { CSS Plugin:

      (7:3) Syntax Error: Unclosed block

        5 |   background-image: url('/assets/file.jpg');
        6 |   /* background-image: url('../../file.png'); */
      > 7 | } {
          |   ^
        8 |
        9 |

          at postcss.process.then.catch (/Users/Cini/Github/Webpack/plugins/css-webpack-plugin/src/lib/worker/workers/postcss.js:22:14)
          at <anonymous>
          at runMicrotasksCallback (internal/process/next_tick.js:115:5)
          at process._tickCallback (internal/process/next_tick.js:150:11) type: 'SyntaxError', '$error': '$error', name: 'CSS Plugin' },
        { CSS Plugin:

      (5:3) Syntax Error: Unclosed block

        3 |   background-image: url('/assets/file.jpg');
        4 |   /* background-image: url('../../../file.png'); */
      > 5 | } {
          |   ^
        6 |
        7 |

          at postcss.process.then.catch (/Users/Cini/Github/Webpack/plugins/css-webpack-plugin/src/lib/worker/workers/postcss.js:22:14)
          at <anonymous>
          at runMicrotasksCallback (internal/process/next_tick.js:115:5)
          at process._tickCallback (internal/process/next_tick.js:150:11) type: 'SyntaxError', '$error': '$error', name: 'CSS Plugin' },
        { CSS Plugin:

      (6:3) Syntax Error: Unclosed block

        4 |   background-image: url('/assets/file.jpg');
        5 |   /* background-image: url('../../../file.png'); */
      > 6 | } {
          |   ^
        7 |
        8 |

          at postcss.process.then.catch (/Users/Cini/Github/Webpack/plugins/css-webpack-plugin/src/lib/worker/workers/postcss.js:22:14)
          at <anonymous>
          at runMicrotasksCallback (internal/process/next_tick.js:115:5)
          at process._tickCallback (internal/process/next_tick.js:150:11) type: 'SyntaxError', '$error': '$error', name: 'CSS Plugin' },
        { CSS Plugin:

      (6:3) Syntax Error: Unclosed block

        4 |   background-image: url('/assets/file.jpg');
        5 |   /* background-image: url('../../../../file.png'); */
      > 6 | } {
          |   ^
        7 |
        8 |

          at postcss.process.then.catch (/Users/Cini/Github/Webpack/plugins/css-webpack-plugin/src/lib/worker/workers/postcss.js:22:14)
          at <anonymous>
          at runMicrotasksCallback (internal/process/next_tick.js:115:5)
          at process._tickCallback (internal/process/next_tick.js:150:11) type: 'SyntaxError', '$error': '$error', name: 'CSS Plugin' } ]
```

![workererror](https://user-images.githubusercontent.com/5419992/36282026-06d51c52-12a0-11e8-9070-70501f7e7a93.png)

Reply:
@michael-ciniawsky `uglify-es` return only one error

Reply:
`postcss` also returns one `{Error}` :). I think this is the part for `{Errors}` I changed based on the worker implementation I initially took from `uglifyjs-webpack-plugin`


```js
const errors = [];
const results = [];

const next = (idx, result) => {
  tasks.pop()

  if (result.err) {
    errors[idx] = result.err;
  } else {
    results[idx] = result;
  }

  if (!tasks.length) {
    cb(errors, results);
  }
};
```